### PR TITLE
[feature-wip](unique-key-merge-on-write) update bitmap after compaction, DSIP-018

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

see DISP-018: https://cwiki.apache.org/confluence/display/DORIS/DSIP-018%3A+Support+Merge-On-Write+implementation+for+UNIQUE+KEY+data+model
This patch is for step 4-2

## Checklist(Required)

1. Type of your changes:
    - [ ] Improvement
    - [ ] Fix
    - [ ] Feature-WIP
    - [ ] Feature
    - [ ] Doc
    - [ ] Refator
    - [ ] Others: 
2. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]
LGTM

**Review by zhannngchen — COMMENTED** [review]
@liaoxin01  you seems ignored the code of clear stale rowset's bitmap when stable rowset is deleted.

File: be/src/olap/tablet_meta.cpp [hunk 1]
```diff
@@ -734,6 +734,45 @@ Status TabletMeta::set_partition_id(int64_t partition_id) {
     return Status::OK();
 }
 
+// update dst rowset delete bitmap for unique key merge on write table.
+// we take a delete bitmap's snapshot of origin rowset, but the
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply be/src/olap/tablet_meta.cpp]: fixed

File: be/src/olap/tablet_meta.cpp [hunk 2]
```diff
@@ -734,6 +734,45 @@ Status TabletMeta::set_partition_id(int64_t partition_id) {
...
+// detele bitmap of origin rowset may update when compaction.
+// some key deleted on compaction will exist in dst rowset.
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply be/src/olap/tablet_meta.cpp]: fixed

File: be/src/olap/tablet_meta.cpp [hunk 3]
```diff
@@ -734,6 +736,46 @@ Status TabletMeta::set_partition_id(int64_t partition_id) {
     return Status::OK();
 }
 
+// update dst rowset delete bitmap for unique key merge on write table.
+// we take a delete bitmap's snapshot of origin rowset at the beginning
```
Comment (zhannngchen) [inline be/src/olap/tablet_meta.cpp:740]: 改完后又看了一遍，有点啰嗦，精简了一下😂
We take a delete bitmap's snapshot of origin rowset at the beginning of compaction, some keys of origin rowsets might be deleted during compaction, but exist in dest rowset. so we need to update the bitmap of dest rowset after compaction.

Comment (liaoxin01) [reply be/src/olap/tablet_meta.cpp]: Good suggestion.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**